modules/dnn/CMakeLists.txt

```diff
@@ -68,7 +71,7 @@ ocv_source_group("Src\\protobuf" FILES ${Protobuf_SRCS} ${Protobuf_HDRS})
 ocv_module_include_directories(include ${Protobuf_INCLUDE_DIRS})
 
 ocv_glob_module_sources(${Protobuf_SRCS} ${Protobuf_HDRS} ${CBLAS_H_PROXY_PATH})
-ocv_create_module(${Protobuf_LIBRARIES} ${LAPACK_LIBRARIES})
+ocv_create_module(${Protobuf_LIBRARIES} ${LAPACK_LIBRARIES} ${OPENCL_LIBRARIES})
```

Comment:
In OpenCV there is T-API that loads OpenCL runtime dynamically. You should not link OpenCL explicitly. Everything, including vendor extensions, is loaded dynamically.